### PR TITLE
[CMS-327] Add linting to pipeline

### PR DESCRIPTION
**Pull request by joejulian**

### CODE REVIEW

**Comment by joejulian** [comment]
This will fail CI as there are linting errors that will need merged before this can be rebased and pass.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -15,3 +15,28 @@ info.
...
+  itself.
+  Follow the link for its documentation.
+- yaml: The yaml linter,
+  [yamllint](https://github.com/adrienverge/yamllint) is packaged for most
+  distros.
+  Installation instructions and documentation can be found at
+  [yamllint.readthedocs.io](https://yamllint.readthedocs.io/).
+- markdown: The markdown linter,
+  [markdownlint-cli](https://github.com/DavidAnson/markdownlint) needs node.js
+  and npm installed to install it.
```
Comment (guineveresaenger) [inline CONTRIBUTING.md]: curious: are we planning to add linting to CI?

Comment (guineveresaenger) [reply CONTRIBUTING.md]: ah, nevermind. was not looking at all the changes

File: Makefile [hunk 2]
```diff
@@ -69,6 +69,16 @@ fmt:
 vet:
 	go vet ./...
 
+golint:
+	@command -v golangci-lint > /dev/null | go get -u github.com/golangci/golangci-lint/cmd/golangci-lint
+	golangci-lint run
```
Comment (guineveresaenger) [inline Makefile]: what golang linter is this? are we all collectively over the gometalinter now?

Comment (guineveresaenger) [reply Makefile]: huh. apparently the answer is yes. Nice!

**Comment by guineveresaenger** [comment]
/lgtm 

I like the shiny new linter. https://github.com/golangci/golangci-lint

**Comment by guineveresaenger** [comment]
@joejulian could you please rebase?

**Pull request merged**